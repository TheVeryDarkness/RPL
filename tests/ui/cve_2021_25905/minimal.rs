//@ revisions: inline regular
//@[inline] compile-flags: -Z inline-mir=true
//@[regular] compile-flags: -Z inline-mir=false

// #[rpl::dump_mir(dump_cfg, dump_ddg)]
fn main() {
    let mut buf: Vec<u8> = Vec::new();
    let b = buf.len();
    let buf = unsafe {
        std::slice::from_raw_parts_mut(buf.as_mut_ptr().offset(b as isize), buf.capacity() - b)
        //~^ ERROR: it violates the precondition of `std::slice::from_raw_parts_mut` to create a slice from uninitialized data
    };
}
