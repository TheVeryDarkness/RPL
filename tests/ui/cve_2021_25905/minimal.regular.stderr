error: it violates the precondition of `std::slice::from_raw_parts_mut` to create a slice from uninitialized data
  --> tests/ui/cve_2021_25905/minimal.rs:10:9
   |
LL |     let mut buf: Vec<u8> = Vec::new();
   |                            ---------- `std::vec::Vec` created but not initialized
LL |     let b = buf.len();
   |             --------- slice created with this length
LL |     let buf = unsafe {
LL |         std::slice::from_raw_parts_mut(buf.as_mut_ptr().offset(b as isize), buf.capacity() - b)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^----------------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                                        |
   |                                        slice created with this pointer
   |
   = help: See https://doc.rust-lang.org/std/slice/fn.std::slice::from_raw_parts_mut.html

error: aborting due to 1 previous error

